packages/cli/src/oclif/commands/canary/create.js

```diff
@@ -0,0 +1,77 @@
+const ZapierBaseCommand = require('../../ZapierBaseCommand');
+const { createCanary, listCanaries } = require('../../../utils/api');
+const { buildFlags } = require('../../buildFlags');
+const { flags } = require('@oclif/command');
+
+class CanaryCreateCommand extends ZapierBaseCommand {
+  async perform() {
+    const { flags } = this.parse(CanaryCreateCommand);
+    const { versionFrom, versionTo, percent, duration } = flags;
+
+    this.validateVersions(versionFrom, versionTo);
+    this.validatePercent(percent);
+    this.validateDuration(duration);
+
+    const existingCanary = await this.findExistingCanary(versionFrom, versionTo);
+    if (existingCanary) {
+      const secondsRemaining = existingCanary.until_timestamp - Math.floor(Date.now() / 1000);
+      this.log(`A canary deployment already exists from version ${versionFrom} to version ${versionTo}, there are ${secondsRemaining} seconds remaining`);
+      return;
+    }
+
+    this.startSpinner(`Creating canary deployment
+    - From version: ${versionFrom}
+    - To version: ${versionTo}
+    - Percent: ${percent}%
+    - Duration: ${duration} seconds`
+    );
+
+    await createCanary(versionFrom, versionTo, percent, duration);
+
+    this.stopSpinner();
+    this.log('Canary deployment created successfully.');
+  }
+
+  async findExistingCanary(versionFrom, versionTo) {
+    const activeCanaries = await listCanaries();
+    return activeCanaries.objects.find(c => c.from_version === versionFrom && c.to_version === versionTo) || null;
+  }
+
+  validateVersions(versionFrom, versionTo) {
+    this.throwForInvalidVersion(versionFrom);
+    this.throwForInvalidVersion(versionTo);
+  }
+
+  validatePercent(percent) {
+    if (!percent || percent < 0 || percent > 100) {
+      this.error('Percent must be between 0 and 100');
+    }
+  }
+
+  validateDuration(duration) {
+    if (!duration || duration <= 30 || duration > 24 * 60 * 60) {
+      this.error('Duration must be a positive number between 30 and 86400');
+    }
+  }
+}
+
+CanaryCreateCommand.flags = buildFlags({
+  commandFlags: {
+    versionFrom: flags.string({char: 'f', description: 'Version to route traffic from', required: true}),
+    versionTo: flags.string({char: 't', description: 'Version to canary traffic to', required: true}),
+    percent: flags.integer({char: 'p', description: 'Percent of traffic to route to new version', required: true}),
+    duration: flags.integer({char: 'd', description: 'Duration of the canary in seconds', required: true}),
+  },
+  opts: {
+    format: true
+  }
+});
+CanaryCreateCommand.description =
+  'Create a new canary deployment, diverting a specified percentage of traffic from one version to another for a specified duration.'
```

Comment:
I'd be a little bit more explicit here so it is clear to users what is going on.
```
Create a new canary deployment, diverting a specified percentage of traffic from one version to another for a specified duration. 

Only one canary can be active at the same time. You can run `zapier canary:list` to check. If you would like to create a new canary with different parameters, you can wait for the canary to finish, or delete it using `zapier canary:delete X Y`. 

Note: this is similar to `zapier migrate` but different in that this is temporary and will "revert" the changes once the specified duration is expired. 
```

Reply:
🤔 I want to say something about breaking changes as well (1.X.Y to 2.X.Y) and advise against it, or to block it from letting the user do it entirely, what do you think?

Reply:
sounds good to me